### PR TITLE
gjs: 1.78.1 -> 1.78.3

### PR DESCRIPTION
**Pull request by trofi**
Changes:
- https://gitlab.gnome.org/GNOME/gjs/-/blob/1.78.2/NEWS
- https://gitlab.gnome.org/GNOME/gjs/-/blob/1.78.3/NEWS

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by amaxine** [comment]
@ofborg build gjs gjs.tests

**Comment by amaxine** [comment]
Tests don't work, but it seems like tests also currently don't work on master. Would be nice to fix that.

**Comment by trofi** [comment]
Yeah, looks like `Xorg` fails to start for `DRI` initialization problems:

```
vm-test-run-gjs> machine # #9  0x00007f9843d6a884 u_pipe_screen_get_param_defaults (kms_swrast_dri.so + 0xc80884)
vm-test-run-gjs> machine # #10 0x00007f984319eb50 dri2_init_screen_extensions (kms_swrast_dri.so + 0xb4b50)
vm-test-run-gjs> machine # #11 0x00007f984319ef70 dri_swrast_kms_init_screen (kms_swrast_dri.so + 0xb4f70)
```

**Comment by trofi** [comment]
Bisected `gjs` test suite regression down to `mesa` update in be9351cb7cbd712c20d35282226288f7321809b8 `mesa: 23.3.1 -> 23.3.2`.

Reverting be9351cb7cbd712c20d35282226288f7321809b8 `mesa: 23.3.1 -> 23.3.2` restores the test.

**Comment by trofi** [comment]
All the test blockers should be fixed now. Rebased against current `staging`.

**Comment by amaxine** [comment]
https://gitlab.gnome.org/GNOME/gjs/-/blob/1.78.3/NEWS fixes a major regression, assuming it doesn't break anything let's update before merging.

**Comment by trofi** [comment]
Updated to 1.78.3

**Review by amaxine — APPROVED** [review]
Thanks. `gjs` builds fine, I'll merge after my local build of tests finishes (been going for ~5 hours so far).

**Comment by amaxine** [comment]
Just finished, all good!

**Pull request merged**
